admin.py

```diff
@@ -0,0 +1,301 @@
+# -*- coding: utf-8 -*-
+""" Trends Earth plugin admin operations,
+    including packaging, deployment and GitHub releases.
+"""
+
+import datetime as dt
+import shlex
+import shutil
+import subprocess
+import typing
+import zipfile
+from dataclasses import dataclass
+from functools import partial
+from pathlib import Path
+
+import httpx
+import typer
+
+LOCAL_ROOT_DIR = Path(__file__).parent.resolve()
+SRC_NAME = "trends.earth"
+PLUGIN_PACKAGE = "LDMP"
+
+PACKAGE_NAME = SRC_NAME.replace("_", "")
+ICON_PATH = "resources/img/icons/icon.png"
+
+TEST_FILES = [
+    "docker-compose.yml",
+    "scripts",
+    "test",
+    "test_suite.py",
+]
+
+PLUGIN_FILES = [
+    "__init__.py",
+    "LDMP",
+    "metadata.txt",
+    "LICENSE" "README.html",
+    "README.md",
+]
+app = typer.Typer()
+
+
+@dataclass
+class GithubRelease:
+    """
+    Class for defining plugin releases details.
+    """
+
+    pre_release: bool
+    tag_name: str
+    url: str
+    published_at: dt.datetime
+
+
+@app.callback()
+def main(context: typer.Context, verbose: bool = False, qgis_profile: str = "default"):
+    """Performs various development-oriented tasks for this plugin
+    :param context: Application context
+    :type context: typer.Context
+    :param verbose: Boolean value to whether more details should be displayed
+    :type verbose: bool
+    :param qgis_profile: QGIS user profile to be used when operating in
+            QGIS application
+    :type qgis_profile: str
+    """
+    context.obj = {
+        "verbose": verbose,
+        "qgis_profile": qgis_profile,
+    }
+
+
+@app.command()
+def generate_zip(
```

Comment:
The plugin uses a number of dependencies for which code to generate a zipfile that packages them with it is now located in the invoke tasks in `tasks.py`. We'll either need to make this function also handle packaging those dependencies, or else borrow from the various development-related tasks that we already have code for in `tasks.py` to make sure this is all handled correctly.

Reply:
@azvoleff sure, I have removed the `admin.py` and will open another PR for it. I'm keeping this PR focused on only running the tests on the CI